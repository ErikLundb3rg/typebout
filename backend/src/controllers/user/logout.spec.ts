/* eslint-disable @typescript-eslint/no-explicit-any */
import { logout } from './index';

describe('Logout Controller', () => {
  let req: any = {};
  let res: any = {};

  beforeEach(() => {
    req = {};
    res = {};
  });

  test('signs out when cookie is present', async () => {
    req = {
      cookies: {
        jid: 'some-cookie'
      }
    };

    res.clearCookie = jest.fn();

    expect(await logout(req, res)).toMatchSnapshot();
    expect(res.clearCookie).toHaveBeenCalled();
  });

  test('fails to sign out when cookie is present', async () => {
    req = {
      cookies: {}
    };
    expect(await logout(req, res)).toMatchSnapshot();
  });
});
