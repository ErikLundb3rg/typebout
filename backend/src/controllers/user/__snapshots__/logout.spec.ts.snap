// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Controller fails to sign out when cookie is present 1`] = `
{
  "data": null,
  "message": "Could not logout, perhaps you are not logged in?",
  "ok": false,
  "status": 400,
}
`;

exports[`Logout Controller signs out when cookie is present 1`] = `
{
  "data": null,
  "message": "Successfully logged out",
  "ok": true,
  "status": 200,
}
`;
